### PR TITLE
 Inequality constraints for Unknowns 

### PR DESCRIPTION
**Pull request by iraikov**
This pull request provides initial support for inequality constraints for Unknowns (issue 32). The dassl backend is at this point configured to check for constraints only in the initial values, as it seems to be having problems with constraints during the simulation.

### CODE REVIEW

**Comment by tshort** [comment]
It'd be nice to have an example that gets run during tests or a specific test case to include these. 

Still on the todo list is sundials support. Also, `initialize!` in src/utils.jl should include these constraints in initial value calculations.

File: src/main.jl [hunk 1]
```diff
@@ -176,6 +176,23 @@ type DefaultUnknown <: UnknownCategory
 end
 
 @doc """
+Categories of constraints on Unknowns; used to create positive, negative, etc., constraints.
+""" ->
+abstract UnknownConstraint
+
+
+type Normal <: UnknownConstraint
```
Comment (tshort) [inline src/main.jl]: These all need documentation strings because they are exported (a oneliner is fine).

**Comment by iraikov** [comment]
Sundials is fully supported; it is dassl that is problematic. I will add the doc strings and look into initialize.

**Comment by tshort** [comment]
Sorry, I missed the Sundials code and misinterpreted your earlier comments on that. Does that mean that Sundials performs fine during integration?

Thanks for doing this. It's a nice feature to have.

**Comment by iraikov** [comment]
No problem, I wasn't very clear myself. Yes, the Sundials backend performs
perfectly fine with or without constraints on several neural examples I
tried. I will add non-negative constraints on the neural/hh,jl example, and
if I think of some other simple example I will add that too.

**Comment by iraikov** [comment]
By the way, it seems that JuMP.@addConstraint does not like constraints of the form y[i] > 0 or y[i] < 0. It looks like it only supports >= <= and ==.

**Comment by tshort** [comment]
I'm good with this whenever you want to merge, @iraikov!

**Pull request merged**